pandas/core/indexes/base.py

```diff
@@ -5371,11 +5371,13 @@ def _cmp_method(self, other, op):
             if len(self) != len(other):
                 raise ValueError("Lengths must match to compare")
 
-        if is_object_dtype(self.dtype) and isinstance(other, ABCCategorical):
-            left = type(other)(self._values, dtype=other.dtype)
-            return op(left, other)
-        elif is_object_dtype(self.dtype) and isinstance(other, ExtensionArray):
-            # e.g. PeriodArray
+        if not isinstance(other, ABCMultiIndex):
+            other = extract_array(other, extract_numpy=True)
```

Comment:
what happens if you have extract_array handle the MI case?

Reply:
extract_array calls MultiIndex.array which raises ValueError

Reply:
not sure i follow. i am suggesting that you handle the MultiIndex case *inside* `extract_array`, e.g.

```
if isinstance(obj, MI): 
   return np.array(obj)
```

Reply:
the this becomes simple here
```
other = extract_array(....)
```

Reply:
extract_array is supposed to be cheep, definitely shouldnt construct a new object array.

i considered adding a mi_safe keyword to return unchanged instead of raising, decided that was a bigger decision than i wanted to make for this PR

Reply:
i c, ok this is definitely kludgy that we have to special case MI

Reply:
for sure, will make a dedicated PR for that after this.